github/commitstatus.go

```diff
@@ -178,6 +182,12 @@ func httpRequestDo(req *http.Request) (httpResponse, error) {
 	response.rateLimitRemaining, _ = strconv.Atoi(resp.Header.Get("x-ratelimit-remaining"))
 	rateLimitReset, _ := strconv.Atoi(resp.Header.Get("x-ratelimit-reset"))
 	response.rateLimitReset = time.Unix(int64(rateLimitReset), 0)
+
+	// time.RFC1123 is the format of the Github date header
+	// example: Date:[ Mon, 02 Jan 2006 15:04:05 MST ]
+	date, _ := time.Parse(time.RFC1123, resp.Header.Get("date"))
```

Comment:
So the date header is always reported and its in `Mon, 02 Jan 2006 15:04:05 MST` format. I can't find anything about this header in Github documentation so I am worried with this a bit. Hence, on L112 I added `req.Header.Set("X-GitHub-Api-Version", "2022-11-28")` to at least try to ensure this doesn't start to break if there is some changes from Github side.

Reply:
I'm not surprised you don't see it in the GitHub doc because this is a standard HTTP header. See https://datatracker.ietf.org/doc/html/rfc2616#section-14.18.

Reply:
To add to Olivier's comment, if an header doesn't begin with "X-" AND the web server is well-behaved, for sure the header is a standard header and so is defined in an RFC.

If an header begins with "X-", then it depends. It can be something invented by the specific service (so in this case GtiHub) OR it could be still a standard header.

A very good starting point is https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

Reply:
I added the link as comment.